lib/plugins/aws/deployFunction.js

```diff
@@ -91,14 +114,40 @@ class AwsDeployFunction {
   }
 
   async callUpdateFunctionConfiguration(params) {
-    await this.provider.request('Lambda', 'updateFunctionConfiguration', params);
+    const startTime = Date.now();
+
+    const callWithRetry = async () => {
+      try {
+        await this.provider.request('Lambda', 'updateFunctionConfiguration', params);
+      } catch (err) {
+        const didOneMinutePass = Date.now() - startTime > 60 * 1000;
+        if (didOneMinutePass) {
+          throw new ServerlessError(
+            'Retry timed out. Please try to deploy your function once again.'
+          );
+        }
```

Comment:
I think this belongs within below `if` condition (otherwise we risk hiding an error that anyway was supposed to be exposed)

Reply:
good call :+1: